docs/guides/serverless-shared-migration-guide.md

```diff
@@ -0,0 +1,242 @@
+        termination_protection_enabled = true
+    }
+    ```
+2. Run `terraform apply` to create the new resource.
+3. Migrate data from your Shared-tier cluster to the Flex cluster using `mongodump` and `mongostore`.
+
+    Please see the following guide on how to retrieve data from one cluster and store it in another cluster: [Convert a Serverless Instance to a Dedicated Cluster](https://www.mongodb.com/docs/atlas/tutorial/convert-serverless-to-dedicated/)
+
+    Verify that your data is present within the Flex cluster before proceeding.
+4. Delete the Shared-tier cluster by running a destroy command against it.
+    
+    For *mongodbatlas_advanced_cluster*:
+
+    `terraform destroy -target=mongodbatlas_advanced_cluster.this`
+
+    For *mongodbatlas_cluster*:
+
+    `terraform destroy -target=mongodbatlas_cluster.this`
+
+ 5. Remove the resource block for the Shared-tier cluster from your `.tf` file.
+
+### Post-Autoconversion Migration Procedure
+
+Shared-tier clusters will automatically convert in March 2025 to Flex clusters in Atlas, retaining all data. 
+
+The following steps resolve the configuration drift in Terraform and does not affect the underlying cluster infrastructure:
+
+1. Find the import IDs of the Flex clusters: `{PROJECT_ID}-{CLUSTER_NAME}`, such as `664619d870c247237f4b86a6-flexClusterName`
+2. Add an import block per cluster to one of your `.tf` files:
+    ```terraform
+    import {
+    to = mongodbatlas_flex_cluster.this
+    id = "664619d870c247237f4b86a6-flexClusterName" # from step 1
+    }
+    ```
+3. Run `terraform plan -generate-config-out=flex_cluster.tf`. This should generate a `flex_cluster.tf` file with your Flex cluster in it.
+4. Run `terraform apply`. You should see the resource(s) imported: `Apply complete! Resources: 1 imported, 0 added, 0 changed, 0 destroyed.`
+5. Re-use existing [Terraform expressions](https://developer.hashicorp.com/terraform/language/expressions). All fields in the generated configuration have static values. Look in your previous configuration for:
+   - variables, for example: `var.project_id`
+   - Terraform keywords, for example: `for_each`, `count`, and `depends_on`
+6. Update the references from your previous cluster resource: `mongodbatlas_advanced_cluster.this.X` or `mongodbatlas_cluster.this.X` to the new `mongodbatlas_flex_cluster.this.X`.
+7. Update any shared-tier data source blocks to refer to `mongodbatlas_flex_cluster`.
+8. Replace your existing clusters with the ones from `flex_cluster.tf` and run 
+
+    `terraform state rm mongodbatlas_advanced_cluster.this`
+
+    or `terraform state rm mongodbatlas_cluster.this`. 
+
+    Without this step, Terraform creates a plan to delete your existing cluster.
+
+9.  Remove the import block created in step 2.
+10.  Re-run `terraform apply` to ensure you have no planned changes: `No changes. Your infrastructure matches the configuration.` 
+
+## From Serverless to Free 
+
+### Pre-Autoconversion Migration Procedure
+
```

Comment:
can we also add a note to do this only if you've verified your serverless cluster can fit into a free cluster?

Reply:
Added requirements for all three Serverless migration guides 👍